src/west/_bootstrap/main.py

```diff
@@ -196,44 +230,118 @@ def init_bootstrap(directory, args):
     # Clone the west source code and the manifest into west/. Git will create
     # the west/ directory if it does not exist.
 
-    clone('west repository', args.west_url, args.west_rev,
+    clone('west repository', west_url, west_rev,
           os.path.join(directory, WEST_DIR, WEST))
 
-    clone('manifest repository', args.manifest_url, args.manifest_rev,
+    clone('manifest repository', manifest_url, manifest_rev,
           os.path.join(directory, WEST_DIR, MANIFEST))
 
     # Create an initial configuration file
 
-    config = configparser.ConfigParser()
+    config_path = os.path.join(directory, WEST_DIR, 'config')
+    update_conf(config_path, west_url, west_rev, manifest_url, manifest_rev)
+    print('=== Initial configuration written to {} ==='.format(config_path))
 
-    config['west'] = {
-        'remote': args.west_url,
-        'revision': args.west_rev
-    }
+    # Create a dotfile to mark the installation. Hide it on Windows.
 
-    config['manifest'] = {
-        'remote': args.manifest_url,
-        'revision': args.manifest_rev
-    }
+    with open(os.path.join(directory, WEST_DIR, WEST_MARKER), 'w') as f:
+        hide_file(f.name)
 
-    config_path = os.path.join(directory, WEST_DIR, 'config')
+    print('=== West initialized ===')
+
+
+def reinit(config_path, args):
+    '''
+    Reinitialize an existing installation.
+
+    This updates the west/config configuration file, and optionally resets the
+    manifest, west, and project repositories to the new revision.
+    '''
+    if args.base_url:
+        if args.west_url or args.manifest_url:
+            sys.exit('fatal error: -b is incompatible with -m and -w')
+        west_url = args.base_url.rstrip('/') + '/west'
+        manifest_url = args.base_url.rstrip('/') + '/manifest'
+    else:
+        west_url = args.west_url
+        manifest_url = args.manifest_url
+
+    if not (west_url or args.west_rev or manifest_url or args.manifest_rev):
+        sys.exit('West already initialized. Please pass any settings you '
+                 'want to change.')
+
+    update_conf(config_path, west_url, args.west_rev, manifest_url,
+                args.manifest_rev)
+
+    print('=== Updated configuration written to {} ==='.format(config_path))
+
+    if args.reset:
+        cmd = ['update']
```

Comment:
@ulfalizer I suggest to change this into:
```
cmd = ['--zephyr-base', 'none', 'update']
```
This will tell `west` when called for re-init, not to parse the manifest.
As this is a re-init, the Zephyr base is not used, and thus `none` is easy understandable.

It will also allow you to switch back, in case the manifest is not present in the revision, as described in #98.
That way you can always recover with a `west init`

You can add it to this PR, or I can make a PR with the suggestion, once this PR is merged.

Note: we could consider to reserve `none` for zephyr base